FLEDGE_Key_Value_Server_trust_model.md

```diff
@@ -0,0 +1,171 @@
+*   Data in transit:
+    *   Client devices will pre-load and periodically refresh the public encryption key signed by a KMS that the client devices recognize.
+    *   The KMS only grants the private key to successfully attested servers. 
+*   Data at rest:
+    *   The server will never do per-request lookups to persistent storage for the key value data. Techniques such as pre-loading the dataset should be used.
+*   Auditability:
+    *   Open source implementations of the key/value service ensure that these systems are publicly accessible and can be inspected by a broad set of stakeholders.
+
+
+
+![alt_text](assets/fledge_key_value_server.png "FLEDGE Key Value Server system diagram")
+
+
+
+## Threats
+
+Ideally, the key-value server would never get access to any information that could be used to identify a specific client or user. However, in practice, request timestamps and other necessary metadata included in lookup requests make this idealized system impractical. 
+
+The main threats that we’re concerned with are:
+
+
+
+1. That the timestamp of the request could be used to correlate the lookup keys with the request of the top-level page and thus identify a user. This is mitigated by limiting the logs that are written.
+2. During the FLEDGE auction, a client asks a key/value server about an assortment of keys that were previously stored on the client — perhaps including keys that were stored in multiple different contexts (different sites or different apps, via [Shared Storage](https://github.com/pythagoraskitty/shared-storage)).  That set of keys, therefore, risks leaking some information about the client's activity.  The privacy goal of key/value server design is to give the client confidence that the set of keys cannot be used for tracking or profiling purposes.  This threat is mitigated by the TEE protections to allow only pre-approved code.
+    1. _Note that the values being stored in the key/value service are loaded by adtechs and expected to be publicly visible to clients. These are therefore not confidential._
+3. The user’s IP address could be used for fingerprinting. This is mitigated by the logging requirements and TEE protections to allow only pre-approved code.
+    2. We also have the option to route the traffic to these servers through [Gnatcatcher](https://github.com/bslassey/ip-blindness) to mask the user’s IP address.
+
+
+### Side effects
+
+There are a number of ways that visible side effects of request handling are possible in general with servers. Here’s how we plan to mitigate those effects:
+
+
+
+1. _Monitoring metrics_ - Only noised aggregate metrics will be available for monitoring and alerting.  These will be aggregated to at least k size.
+    1. For example, counting the approximate number of failed requests in the past n minutes is likely to be fine but doing so at millisecond granularity is not.
```

Comment:
Formatting for nested lists is wrong — the "1." of the nested lists needs to be indented to be under the first character of the list item.  See details here: 

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#nested-lists

Reply:
Thanks, fixed